packages/server-wallet/src/__test__/crash-tolerance/crash-tolerance.test.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  CreateChannelParams,
+  Participant,
+  Allocation,
+  CloseChannelParams,
+} from '@statechannels/client-api-schema';
+import {makeDestination} from '@statechannels/wallet-core';
+import {BigNumber, ethers} from 'ethers';
+
+import {defaultConfig as processEnvConfig} from '../../config';
+import {Wallet} from '../../wallet';
+import {getChannelResultFor, getPayloadFor} from '../test-helpers';
+
+const a = new Wallet({...processEnvConfig, postgresDBName: 'TEST_A'});
+let b = new Wallet({...processEnvConfig, postgresDBName: 'TEST_B'}); // Wallet that will "crash"
+
+let channelId: string;
+let participantA: Participant;
+let participantB: Participant;
+
+beforeAll(async () => {
+  await a.dbAdmin().createDB();
+  await b.dbAdmin().createDB();
+  await Promise.all([a.dbAdmin().migrateDB(), b.dbAdmin().migrateDB()]);
+
+  participantA = {
+    signingAddress: await a.getSigningAddress(),
+    participantId: 'a',
+    destination: makeDestination(
+      '0xaaaa000000000000000000000000000000000000000000000000000000000001'
+    ),
+  };
+  participantB = {
+    signingAddress: await b.getSigningAddress(),
+    participantId: 'b',
+    destination: makeDestination(
+      '0xbbbb000000000000000000000000000000000000000000000000000000000002'
+    ),
+  };
+});
+afterAll(async () => {
+  // Don't destroy b here, it get's destroyed midway through test
+  await Promise.all([a.destroy(), b.destroy()]);
+  await a.dbAdmin().dropDB();
+  await b.dbAdmin().dropDB(); // Only need to drop the db once as it is shared with b
+});
+
+it('Create a fake-funded channel between two wallets, of which one crashes midway through ', async () => {
+  const allocation: Allocation = {
+    allocationItems: [
+      {destination: participantA.destination, amount: BigNumber.from(1).toHexString()},
+      {destination: participantB.destination, amount: BigNumber.from(1).toHexString()},
+    ],
+    token: '0x00', // must be even length
+  };
+
+  const createChannelParams: CreateChannelParams = {
+    participants: [participantA, participantB],
+    allocations: [allocation],
+    appDefinition: ethers.constants.AddressZero,
+    appData: '0x00', // must be even length
+    fundingStrategy: 'Direct',
+  };
+
+  //        A <> B
+  // PreFund0
+  const resultA0 = await a.createChannel(createChannelParams);
+
+  // TODO compute the channelId for a better test
+  channelId = resultA0.channelResults[0].channelId;
```

Comment:
We have this TODO in 3 or 4 files now... maybe we ought to either remove it, add an issue, or do it? :p